torch/csrc/CudaIPCTypes.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+#include <TH/TH.h>
+
+struct CudaIPCReceivedData {
+  CudaIPCReceivedData(std::shared_ptr<void> shared_ptr);
+  std::shared_ptr<void> shared_ptr_;
+};
+
+struct CudaIPCSentData {
+  CudaIPCSentData(at::DataPtr shared_storage_ptr);
+  ~CudaIPCSentData();
+  int64_t get();
+  at::DataPtr original_ptr_; // Original mem allocation
+  at::DataPtr shared_storage_ptr_; // Reference counter shared memory block
+};
+
+// All to be deleted data blocks with non zero reference counter goes there
+struct CudaIPCSentDataLimbo {
+  void collect();
+  ~CudaIPCSentDataLimbo();
+  void add(CudaIPCSentData* shared_block);
+private:
+  std::vector<CudaIPCSentData*> shared_blocks_;
```

Comment:
I see you manually deleting these pointers elsewhere; sounds like a `unique_ptr` is in order here